### PR TITLE
report abuse dropdown link

### PR DESCRIPTION
**Pull request by lboogie04**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Adds a Report Abuse link to the Billboard dropdown that essentially prefills the `reported_url` field with the url the billboard was spotted on along with the `display_ad` ID. 
(i.e. `https://dev.to/suede/build-ai-template-engine-to-create-amazing-content-with-nextjs-and-chatgpt-3f89?display_ad=33` )

Screenshot 2023-03-21 at 1 34 44 AM

## Related Tickets & Documents

- Related Issue #
- Closes [#568](https://github.com/forem/forem-internal-eng/issues/568)

## QA Instructions, Screenshots, Recordings
 1. Create a billboard
 2. set the billboard anywhere(on the home page OR article). Preferably on an article to test various urls.
 3. If the display ad is a billboard, you should see the Report Abuse link in the ad dropdown. 
 4. Clicking the link should direct you to the Report Abuse page with the reported url field populated with the url you just came from with a `display_ad` param attached to the url.

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -82,9 +82,10 @@ def terms
   end
 
   def report_abuse
+    billboard_url = params[:billboard]
```
Comment (jaw6) [inline app/controllers/pages_controller.rb:85]: I think we want the link to take the moderator to somewhere they can view the problematic billboard. What about something like this?

```suggestion
    billboard_url = admin_display_ad_path(params[:billboard]) if params[:billboard].present?
```

File: app/views/shared/_display_ad.html.erb [hunk 2]
```diff
@@ -32,6 +32,15 @@
             <%= I18n.t("display_ad.menu.manage_preferences") %>
           </a>
         </li>
+        <% if Page.exists?(slug: "billboards") %>
+          <hr />
+          <li>
+            <a class="crayons-link crayons-link--block" href="/report-abuse?billboard=<%= request.original_url %>?display_id=<%= display_ad.id %>">
```
Comment (jaw6) [inline app/views/shared/_display_ad.html.erb:38]: I think the double `?` here will cause some hard-to-predict cross-browser behaviors. Based on the controller above, which wants this to pass a `billboard`, I am thinking this wants to be more like:

```suggestion
            <a class="crayons-link crayons-link--block" href="/report-abuse?billboard=<%= display_ad.id %>">
```

**Review by jaw6 — CHANGES_REQUESTED** [review]

**Review by jaw6 — APPROVED** [review]

File: app/views/shared/_display_ad.html.erb [hunk 3]
```diff
@@ -32,6 +32,15 @@
             <%= I18n.t("display_ad.menu.manage_preferences") %>
           </a>
         </li>
+        <% if Page.exists?(slug: "billboards") %>
```
Comment (Ridhwana) [inline app/views/shared/_display_ad.html.erb:35]: Question: Why are we using this if condition to show the element? 

We use it a few lines above too, but in that case its used to ensure that the url that we redirect to exists:

```
 <% if Page.exists?(slug: "billboards") %>
          <li>
            <a class="crayons-link crayons-link--block" href='/billboards'>
              <%= crayons_icon_tag("info", aria_hidden: true, class: "c-btn__icon") %>
              <%= I18n.t("display_ad.menu.what_is_a_billboard") %>
            </a>
          </li>
        <% end %>
```

Comment (Ridhwana) [reply app/views/shared/_display_ad.html.erb:42]: @lboogie04 a display_ad and a billboard are synonymous terms, we've decided to use the word billboard more recently but its still being called display_ad in the code. Eventually we hope to rename it across the codebase and UI. 

`Page.exists?(slug: "billboards")` checks if the information page for billboards exists, but this is not related to whether or not we display a report link on the billboard/display_ad itself. We should still have the ability to report a billboard/display_ad even if there is no information page describing billboards. 

Does this make sense? Let me know if we need to chat more about it, i know that these terms can be a bit confusing.

**Pull request merged**